File: python/docs_code_snippets/knowledge-files-list.py
import time
from opperai import Opper

opper = Opper()

# Setup
kb = opper.knowledge.create(name=f"docs-snippet-{int(time.time())}")
opper.knowledge.upload_file(kb.id, b"test content", filename="document.txt")

# --- docs ---
files = opper.knowledge.list_files(kb.id)
for f in files:
    print(f"{f.original_filename} ({f.size} bytes, status: {f.status})")
# --- /docs ---

# Cleanup
opper.knowledge.delete(kb.id)

File: typescript/docs_code_snippets/knowledge-files-list.ts
import { Opper } from "opperai";

const opper = new Opper();

// setup
const kb = await opper.knowledge.create({ name: `docs-snippet-${Date.now()}` });
const kbId = kb.id;
await opper.knowledge.uploadFile(kbId, new Blob(["test content"], { type: "text/plain" }), {
  filename: "test.txt",
});
// /setup

// --- docs ---
const files = await opper.knowledge.listFiles(kbId);
for (const f of files.data) {
  console.log(`${f.original_filename} (${f.size} bytes, status: ${f.status})`);
}
// --- /docs ---

// cleanup
await opper.knowledge.delete(kbId);
